tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -1831,6 +1833,9 @@ HRESULT DxcCompilerAdapter::WrapCompile(
     pResult->CopyOutputsFromResult(pImplResult);
     pResult->SetStatusAndPrimaryResult(hr, pImplResult->PrimaryOutput());
 
+    if (opts.TimeReport)
+      llvm::TimerGroup::printAll(outStream);
```

Comment:
This should dump the output to the same stream that compiler errors and warnings go out, which I think is what we want because it matches the behavior in Clang.

We could also support mapping to a separate file, which would be clang's `-ftime-report=` option.

Reply:
I would think it would be nicer to have it written to a separate DXC_OUT_KIND stream. Either a new one like `DXC_OUT_FTIME` or maybe `DXC_OUT_REMARKS`. The dxc.exe behavior would be the same (writing that output to stdout/stderr), but the API usage would be nicer.

I don't think that necessarily needs to be done for the first checkin, but we should decide if we want to do that soon before people start relying on a particular behavior. What do you think @tex3d?

In any case we should add a compile test that passes the `-ftime arg and reads the data from `DXC_OUT_ERRORS` output (or the separate output if we want to go that route).

Reply:
I like the idea of a separate output.  Perhaps `DXC_OUT_TIME_REPORT`?  If the option for separate output is `-ftime-report=` in Clang, how do we separate that from a flag that doesn't set an output file for Dxc?  Could we do something here where `-` means print inline to the stderr output for dxc.exe, while placing a filename here would keep the output separate and dump to the specified file?